File: app/src/main/java/com/example/runerrands/room/database/AddressDataBase.java
package com.example.runerrands.room.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.runerrands.room.bean.Address;
import com.example.runerrands.room.dao.AddressDao;


@Database(entities = {Address.class},version = 1,exportSchema = false)
public abstract class AddressDataBase extends RoomDatabase {

    private static AddressDataBase addressDataBase;

    public static synchronized AddressDataBase get(Context context){
        if (addressDataBase == null){
            addressDataBase = Room.databaseBuilder(context.getApplicationContext(),AddressDataBase.class,"address.db")
                    .build();

        }
        return addressDataBase;
    }


    public abstract AddressDao getAddressDao();
}

File: app/src/main/java/com/example/runerrands/room/database/AddressDatabase.kt
package com.example.runerrands.room.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.runerrands.room.bean.Address
import com.example.runerrands.room.dao.AddressDao

@Database(entities = [Address::class],version = 1,exportSchema = false)
abstract class AddressDatabase: RoomDatabase() {

    companion object{
        @Volatile
        private var INSTANCE: AddressDatabase? = null
        fun getInstance(context: Context)
                = INSTANCE ?: synchronized(this){
            INSTANCE ?: buildDatabase(context).also{INSTANCE = it}
        }
        private fun buildDatabase(context: Context) =
            Room.databaseBuilder(context.applicationContext, AddressDatabase::class.java, "address.db")
                .build()
    }

    abstract fun getAddressDao(): AddressDao
}